backend/clap/src/main/java/zpi/ppea/clap/exceptions/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,49 @@
+package zpi.ppea.clap.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import zpi.ppea.clap.security.FirebaseAgent;
+
+import java.util.*;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(UserNotAuthorizedException.class)
+    public ResponseEntity<String> handleUserNotAuthorizedException(UserNotAuthorizedException ex) {
+        return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+                .header(FirebaseAgent.REFRESH_TOKEN_NAME, ex.getRefresh()).body(ex.getMessage());
+    }
+
+    @ExceptionHandler(GrpcDatastoreException.class)
+    public ResponseEntity<String> handleNoAccessToResourceException(GrpcDatastoreException ex) {
+        return ResponseEntity.status(HttpStatus.FORBIDDEN)
+                .header(FirebaseAgent.REFRESH_TOKEN_NAME, ex.getRefresh()).body(ex.getMessage());
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<Map<String, List<String>>> handleValidationErrors(MethodArgumentNotValidException ex) {
+        List<String> errors = new ArrayList<>();
+        for (var messages : Objects.requireNonNull(ex.getDetailMessageArguments()))
+            errors.addAll((Collection<? extends String>) messages);
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                .header(FirebaseAgent.REFRESH_TOKEN_NAME, "false")
+                .body(getErrorsMap(errors));
+    }
+
+    @ExceptionHandler(Exception.class)
```

Comment:
Doesn't this work like catch-all? I don't think we want to do that - for example, we don't want to catch NullPointerExceptions, right? This will quietly hide all those potential NPE's

Reply:
I asked you about 'refresh header' and you said it should be always returned by clap to frontend, so to make it possible I have to provide this 'every exception handler'.

Reply:
That's not the point. We should only handle exceptions that either we explicitly throw, or other declared by dependencies exceptions (like gRPC exceptions or Firebase ones) -- any exception we anticipate to occur should be dealt with by exception handler, unexpected exceptions like NPE's shouldn't happend, but if they do we don't want to hide them.